.grit/patterns/js/replace-zlib-deflate-to-zlib-deflateSync.md

```diff
@@ -0,0 +1,94 @@
+---
+title: Replace `zlib.deflate` ⇒ `zlib.deflateSync`
+---
+
+Creating and using a large number of `zlib` objects simultaneously can cause significant memory fragmentation. It is strongly recommended that the results of compression operations be cached or made synchronous to avoid duplication of effort
+
+- [reference](https://nodejs.org/api/zlib.html#zlib_threadpool_usage_and_performance_considerations)
+
+tags: #fix #best-practice
+
+```grit
+engine marzano(0.1)
+language js
+
+or {
+    `for($a;$b;$c){$body}`,
+    `$arr.forEach(($condition) => {$body})`,
+    `while($condition){$body}`,
+    `do {$body} while($condition)`
+} where {
+    $body <: contains `zlib.deflate` => `zlib.deflateSync`
```

Comment:
The main clause of a pattern should always be the primary thing we're interested in. So in this case it should be inverted: look for a zlib.deflate and *then* check if it is `within` a loop like.

Reply:
updated the suggested changes.